src/wallet/mod.rs

```diff
@@ -837,6 +836,9 @@ where
             if self.is_mine(&drain_output.script_pubkey)? {
                 received += drain_val;
             }
+            // TODO: We should pay attention when adding a new output: this might increase
+            // the lenght of the "number of vouts" parameter by 1 byte, potentially making
+            // our feerate too low
```

Comment:
Just to warn here that the addition of the draining output can increase the size of length field of the vout vector (and any VarInt in general), not by 1 but 2 or 4 bytes, depending on its previous length. More about this [here](https://learnmeabitcoin.com/technical/varint) and [here](https://docs.rs/bitcoin/latest/src/bitcoin/consensus/encode.rs.html#361-401).

Reply:
LGTM. I didn't think in all that, great explanation, thanks!

Reply:
Oh, TIL! I wasn't sure how varint worked. I just updated the comment, thanks 🙏🏻